lib/tree.rb

```diff
@@ -1,64 +1,179 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, value)
     @key = key
-    @value = val
+    @value = value
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: Olog(n) - only need to look at half the tree
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
👍 Since you're doing this recursively, the space complexity is O(log n) if the tree is balanced and O(n) if it is unbalanced.  Same for time complexities.